src/components/MetadataManager.js

```diff
@@ -0,0 +1,224 @@
+
+  constructor(props: MetadataManagerProps) {
+    super(props);
+    this.state = {
+      response: undefined,
+      error: undefined,
+      savedSearchList: [],
+    };
+    (this: any).getSavedSearches = this.getSavedSearches.bind(this);
+    (this: any).saveThisSearch = this.saveThisSearch.bind(this);
+    (this: any).deleteThisSearch = this.deleteThisSearch.bind(this);
+    (this: any).populateSavedSearches = this.populateSavedSearches.bind(this);
+    (this: any).updateSavedSearches = this.updateSavedSearches.bind(this);
+  }
+
+  state: MetadataManagerState;
+
+  getChildContext() {
+    return {
+      metadataManager: this,
+    };
+  }
+
+  // queries the index for the saved searches and sets the response to state
+  getSavedSearches() {
+    const username = AuthUtils.getLoggedInUserId();
+    const searcher = this.context.searcher;
+    if (searcher) {
+      const qr = new SimpleQueryRequest();
+      qr.query = `AND(table:'.savedSearches', username_s:"${username}")`;
+      qr.facets = [];
+      qr.restParams.set('zones', ['.metadata']);
+      qr.workflow = 'search';
+      qr.queryLanguage = 'advanced';
+      searcher.doCustomSearch(qr, (response: QueryResponse | null, error: string | null) => {
+        if (response) {
+          this.populateSavedSearches(response);
+        } else if (error) {
+          this.setState({ error });
+        }
+      });
+    }
+  }
+
+  // populates the response in an array of SavedSearch objects
+  populateSavedSearches(response) {
+    if (response) {
+      const menuItemList = [];
+      response.documents.forEach((doc) => {
+        const ss = new SavedSearch();
+        ss.setProperties(doc);
+        menuItemList.push(ss);
+      });
+      this.setState({ savedSearchList: menuItemList });
+    }
+  }
+
+  // saves a search query from the url and feeds it to a separate index zone, with an optional callback
+  saveThisSearch(callback: () => void) {
+    const { searcher } = this.context;
+    const loggedDateTime = new Date().toISOString();
+    const username = AuthUtils.getLoggedInUserId();
+    const ss = new SavedSearch();
+    ss.id = username.concat(loggedDateTime);
+    ss.title = this.context.searcher.state.query;
+    ss.query = searcher.generateLocationQueryStringFromState(this.context.searcher.state);
+    ss.queryString = searcher.state.query;
+    searcher.search.updateSavedSearches(ss, this.getSavedSearches);
+    callback();
+  }
+
+  deleteThisSearch(id: string) {
+    const ss = new SavedSearch();
+    ss.id = id;
+    const mode = 'DELETE';
+    this.updateSavedSearches(ss, this.getSavedSearches, mode);
+  }
+
+  updateSavedSearches(ss: SavedSearch, callback: () => void, mode: string = 'ADD'): Promise<any> {
+    return new Promise((resolve, reject) => {
+      // Get session
+      const baseUri = this.context.searcher.search.baseUri;
+      const connectUri = `${baseUri}/rest/ingestApi/connect`;
+      fetch(connectUri, { credentials: 'include' })
```

Comment:
This logic should be in the `api` directory instead of inside a component. Avoid putting complex fetching logic in components.